src/hip_memory.cpp

```diff
@@ -1291,25 +1311,48 @@ hipError_t ihipMemcpy3D(const struct hipMemcpy3DParms* p, hipStream_t stream, bo
                 ySize = p->srcPtr.ysize;
                 desc = p->dstArray->desc;
                 dstPtr = p->dstArray->data;
+                hsa_ext_image_data_info_t imageInfo;
+                if(hipTextureType2DLayered == p->dstArray->textureType)
+                    GetImageInfo(HSA_EXT_IMAGE_GEOMETRY_2DA, width, height, 0, desc, imageInfo, depth);
+                else
+                    GetImageInfo(HSA_EXT_IMAGE_GEOMETRY_3D, width, height, depth, desc, imageInfo);
+                dstPitch = imageInfo.size/(height == 0 ? 1:height)/(depth == 0 ? 1:depth);
             } else {
                 depth = p->Depth;
                 height = p->Height;
                 widthInBytes = p->WidthInBytes;
                 width =  p->dstArray->width;
-                desc = hipCreateChannelDesc(32, 0, 0, 0, hipChannelFormatKindSigned);
+                hsa_ext_image_channel_order_t channelOrder;
+                switch(p->dstArray->NumChannels) {
+                    case 1:
+	                    channelOrder = HSA_EXT_IMAGE_CHANNEL_ORDER_R;
+	                    break;
+	                case 2:
+	                    channelOrder = HSA_EXT_IMAGE_CHANNEL_ORDER_RG;
+	                    break;
+	                case 3:
+	                    channelOrder = HSA_EXT_IMAGE_CHANNEL_ORDER_RGB;
+	                    break;
+	                case 4:
+	                    channelOrder = HSA_EXT_IMAGE_CHANNEL_ORDER_RGBA;
+	                    break;
+	                default:
+	                    channelOrder = HSA_EXT_IMAGE_CHANNEL_ORDER_R;
+	                    break;
+                }
+                hsa_ext_image_channel_type_t channelType;
+                e = ihipArrayToImageFormat(p->dstArray->Format,channelType);
                 srcPitch = p->srcPitch;
                 srcPtr = (void*)p->srcHost;
                 ySize = p->srcHeight;
                 dstPtr = p->dstArray->data;
+                hsa_ext_image_data_info_t imageInfo;
+                if(hipTextureType2DLayered == p->dstArray->textureType)
+                    GetImageInfo(HSA_EXT_IMAGE_GEOMETRY_2DA, width, height, 0, channelOrder, channelType, imageInfo, depth);
+                else
+                    GetImageInfo(HSA_EXT_IMAGE_GEOMETRY_3D, width, height, depth, channelOrder, channelType, imageInfo);
+                dstPitch = imageInfo.size/(height == 0 ? 1:height)/(depth == 0 ? 1:depth);
```

Comment:
formatting: missing spaces

Reply:
dstPitch calculation is not part of else.

Reply:
Spaces here, around colon: 
```c++
(height == 0 ? 1:height)/(depth == 0 ? 1:depth);
```